x-pack/plugins/cloud_security_posture/server/routes/configuration/update_rules_configuration.ts

```diff
@@ -4,68 +4,60 @@
+import type { AuthenticatedUser } from '@kbn/security-plugin/common';
 import { unset } from 'lodash';
-import yaml from 'js-yaml';
-
-import { PackagePolicy, PackagePolicyConfigRecord } from '@kbn/fleet-plugin/common';
-import { PackagePolicyServiceInterface } from '@kbn/fleet-plugin/server';
-import { AuthenticatedUser } from '@kbn/security-plugin/common';
+import produce from 'immer';
 import { createCspRuleSearchFilterByPackagePolicy } from '../../../common/utils/helpers';
-
 import type { CspRule, CspRulesConfiguration } from '../../../common/schemas';
 import {
-  CLOUD_SECURITY_POSTURE_PACKAGE_NAME,
   CSP_RULE_SAVED_OBJECT_TYPE,
   UPDATE_RULES_CONFIG_ROUTE_PATH,
 } from '../../../common/constants';
-import { CspRouter } from '../../types';
+import type { CspApiRequestHandlerContext, CspRouter, CspRequestHandler } from '../../types';
 
-export const getPackagePolicy = async (
-  soClient: SavedObjectsClientContract,
-  packagePolicyService: PackagePolicyServiceInterface,
-  packagePolicyId: string
-): Promise<PackagePolicy> => {
-  const packagePolicies = await packagePolicyService.getByIDs(soClient, [packagePolicyId]);
+export type UpdateRulesConfigBodySchema = TypeOf<typeof updateRulesConfigurationBodySchema>;
 
-  // PackagePolicies always contains one element, even when package does not exist
-  if (!packagePolicies || !packagePolicies[0].version) {
-    throw new Error(`Package policy Id '${packagePolicyId}' does not exist`);
-  }
-  if (packagePolicies[0].package?.name !== CLOUD_SECURITY_POSTURE_PACKAGE_NAME) {
-    throw new Error(
-      `Package Policy Id '${packagePolicyId}' is not of type cloud security posture package`
-    );
-  }
+/** Minimal set of fields required for updating a CSP SO rule */
+type SavedObjectRuleUpdatePayload = Pick<CspRule, 'enabled'>;
 
-  return packagePolicies![0];
-};
+/**
+ * Minimal set of fields required for generating
+ * the value of the runtime config key on the package policy vars object
+ */
+export interface PackagePolicyRuleUpdatePayload {
+  enabled: boolean;
+  metadata: { benchmark: { id: string }; rego_rule_id: string };
+}
 
-export const getCspRules = (
-  soClient: SavedObjectsClientContract,
-  packagePolicy: PackagePolicy
-): Promise<SavedObjectsFindResponse<CspRule, unknown>> => {
-  return soClient.find<CspRule>({
-    type: CSP_RULE_SAVED_OBJECT_TYPE,
-    filter: createCspRuleSearchFilterByPackagePolicy({
-      packagePolicyId: packagePolicy.id,
-      policyId: packagePolicy.policy_id,
-    }),
-    searchFields: ['name'],
-    // TODO: research how to get all rules
-    perPage: 10000,
-  });
-};
+const RUNTIME_CFG_FIELDS = ['enabled', 'metadata.benchmark.id', 'metadata.rego_rule_id'];
+
+const updateRulesConfigurationBodySchema = rt.object({
+  /**
+   * CSP integration instance ID
+   */
+  package_policy_id: rt.string(),
+  /**
+   * Rules to update
+   */
+  rules: rt.arrayOf(
+    rt.object({
+      id: rt.string(),
```

Comment:
it's little confusing because it's not the rego rule id

Reply:
added a comment